server/helpers/adapters_tracker.go

```diff
@@ -119,11 +122,31 @@ func (a *AdaptersTracker) DeployAdapter(ctx context.Context, adapter models.Adap
 			return ErrAdapterAdministration(err)
 		}
 
+	case "kubernetes":
+		kubeClient, err := meshkitkube.New([]byte(""))
+		if err != nil {
+			return ErrAdapterAdministration(err)
+		}
+
+		exposedPort, _ := strconv.Atoi(adapter.Location)
+		err = a.applyHelmCharts(kubeClient, adapter.Name, int32(exposedPort), true)
+		if err != nil {
+			return ErrAdapterAdministration(err)
+		}
+
+		svc, err := kubeClient.KubeClient.CoreV1().Services(core.MesheryNamespace).Get(context.Background(), adapter.Name, metav1.GetOptions{})
+		if err != nil {
+			return ErrAdapterAdministration(err)
+		}
+
+		targetIP = svc.Spec.ClusterIP
+
 	// switch to default case if the platform specified is not supported
 	default:
 		return ErrAdapterAdministration(fmt.Errorf("the platform %s is not supported currently. The supported platforms are:\ndocker\nkubernetes", platform))
 	}
 
+	adapter.Location = targetIP + ":" + adapter.Location
```

Comment:
Is this how we have provisioned components of Meshery Operator? There's no need for TargetIP. Use the adapter name instead. Rely on DNS inside of Kubernetes and Docker-based deployments.